dbt_subprojects/dex/macros/models/_project/eulerswap_compatible_trades.sql

```diff
@@ -0,0 +1,215 @@
+    where 1=1
+    {%- if is_incremental() %}
+    and {{ incremental_predicate('evt_block_time') }}
+    {%- endif %}
+
+),
+
+dexs as (
+    select 
+        s.evt_block_time as block_time
+        , s.evt_block_number as block_number 
+        , ep.asset0 as token0 
+        , ep.asset1 as token1
+        , ep.pool as maker -- uniswap pool id 
+        , ep.fee
+        , ep.protocolFee
+        , s.contract_address as instance -- hook
+        , s.contract_address as project_contract_address -- rename for dex trades
+        , ep.eulerAccount 
+        , ep.factory_address
+        , ep.creation_block_time as pool_creation_time 
+        , s.evt_tx_hash as tx_hash
+        , s.evt_index as tx_index 
+    
+        -- EulerSwap Instance event from pool's perspective, so flipping the sign to be user's perspective
+        , case 
+            when s.amount0In != 0 then -1 * cast(s.amount0In as double)
+            else cast(s.amount0Out as double)
+          end as amount0 
+        , case 
+            when s.amount1In != 0 then -1 * cast(s.amount1In as double)
+            else cast(s.amount1Out as double)
+          end as amount1
+        , s.amount0In
+        , s.amount0Out
+        , s.amount1In
+        , s.amount1Out
+        , s.sender -- router 
+        , s.evt_index
+        , t.tx_hash as uni_tx_hash
+    from 
+    {{ eulerswapinstance_evt_swap }} s 
+    left join 
+    {{ eulerswap_pools_created }} ep 
+        on ep.hook = s.contract_address 
+    left join 
+    uni_v4_trades t 
+        on t.block_number = s.evt_block_number 
+        and t.tx_hash = s.evt_tx_hash
+        and t.evt_index = s.evt_index + 1 
+    where 1 = 1 
+    {% if is_incremental() %}
+    and {{ incremental_predicate('s.evt_block_time') }}
+    {% endif %}
+)
+
+select 
+    '{{ blockchain }}' AS blockchain
+    , '{{ project }}' AS project
+    , '{{ version }}' AS version
+    , CAST(date_trunc('month', dexs.block_time) AS date) AS block_month
+    , CAST(date_trunc('day', dexs.block_time) AS date) AS block_date
+    , dexs.block_time
+    , dexs.block_number
+    , case 
+        when amount0Out != 0 then amount0Out
+        else amount1Out
+      end as token_bought_amount_raw 
+    , case 
+        when amount0In != 0 then amount0In
+        else amount1In
+      end as token_sold_amount_raw 
+    , case 
+        when amount0Out != 0 then token0
+        else token1
+      end as token_bought_address
+    , case 
+        when amount0In != 0 then token0
+        else token1
+      end as token_sold_address
+    , dexs.sender as taker
+    , dexs.project_contract_address as maker
+    , dexs.project_contract_address
+    , dexs.tx_hash
+    , dexs.evt_index
+    -- add eulerswap trades items at the end 
+    , dexs.pool_creation_time
+    , dexs.fee 
+    , dexs.protocolFee 
+    , dexs.instance 
+    , dexs.eulerAccount 
+    , dexs.factory_address 
+    , dexs.sender 
+from 
+dexs 
+where uni_tx_hash is not null 
```

Comment:
i think you actually have this backwards, should be
`where uni_tx_hash is null`
which says "this euler tx_hash does not appear in uniswap v4 decoded swap, therefore add"
we want the lookup to return nothing to be a net new row to add

Reply:
ahh sorry yeah, fixed now!